### PR TITLE
feat: external cloudwatch kms and sns topic

### PR DESCRIPTION
**Pull request by j0hnmwangi**
# Description
## Changes made in this PR:

- feat: option to use an external key for cloudwatch logs encryption.
- feat: option to use an external sns topi for autoscaling notifications.
- fix: autoscaling tags variable.

## Features/Fixes/Patches/Changes list:

Please check the [CHANGELOG.md](https://github.com/boldlink/terraform-aws-autoscaling/blob/feat/external-cw-kms/CHANGELOG.md#220---2024-04-15)

### PR Owners Checklist:
* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] Have you checked there aren't open [Pull Requests](../../../pulls) of upstream or parallel code that will cause conflicts?
* [x] Have you updated the `CHANGELOG.md`?
* [x] Have you updated the `README.md`(s)?
* [x] OWNER: Does your submission pass?
    * [ ] Pre-commit (attach logs/print screen to this PR)
    * [ ] Checkov/terrascan (attach logs/print screen to this PR)
    * [ ] Terraform resources examples build/destroy successfully with Terraform 0.14.11 minimum?
* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### PR Reviewers Checklists?
* [ ] Are all your comments/changes resolved?
* [ ] Are you happy with the OWNER checklists and additional information provided?
* [ ] Does your review tests pass?
    * [ ] Terraform resources examples build/destroy successfully with Terraform 0.14.11 minimum?

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,13 +22,17 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
-- feat: Add more options to secrets manager where pem key is store (e.g tags, retention e.t.c).
 - feat: Remove/upgrade usage of deprecated hashicorp/template provider
 - fix: CKV_AWS_356 "Ensure no IAM policies documents allow “*” as a statement’s resource for restrictable actions"
 - fix: CKV2_AWS_57 "Ensure Secrets Manager secrets should have automatic rotation enabled"
 - fix: CKV_AWS_341 "Ensure Launch template should not have a metadata response hop limit greater than 1"
 - fix: CKV_TF_1 "Ensure Terraform module sources use a commit hash"
 
+## [2.2.0] - 2024-04-15
+- feat: option to use an external key for cloudwatch logs encryption
+- feat: option to use an external sns topi for autoscaling notifications
```
Comment (hugoalmeida264) [inline CHANGELOG.md:33]: Typo on topi

File: examples/complete/main.tf [hunk 2]
```diff
@@ -496,6 +513,8 @@ module "spot_one_time" {
   }
 }
 
+## Autoscaling Group with Accelerators example
+
```
Comment (hugoalmeida264) [inline examples/complete/main.tf:517]: Remove /n

File: examples/complete/main.tf [hunk 3]
```diff
@@ -1,25 +1,37 @@
...
+  depends_on               = [aws_launch_template.external]
+  enable_asg_events_notify = true
+  topic_arn                = module.sns_topic.arn
 }
 
+## Complete Autoscaling Group Example
+
 module "ebs_kms" {
   source           = "boldlink/kms/aws"
   version          = "1.1.0"
```
Comment (hugoalmeida264) [inline examples/complete/main.tf:37]: Use the latest kms module version

File: variables.tf [hunk 4]
```diff
@@ -495,6 +489,18 @@ variable "extra_script" {
 ##############
 ###Alarms
 ##############
+variable "enable_asg_events_notify" {
+  description = "Whether to create AutoScaling Group with Notification support, via SNS Topics"
+  type        = bool
+  default     = false
+}
+
+variable "create_asg_sns_topic" {
```
Comment (hugoalmeida264) [inline variables.tf:498]: Do we need this variable or we can use the above to imply the above?

File: variables.tf [hunk 5]
```diff
@@ -518,6 +530,12 @@ variable "sns_kms_master_key_id" {
   default     = "alias/aws/sns"
```
Comment (hugoalmeida264) [inline variables.tf:530]: Is this value for a kms key?

**Review by hugoalmeida264 — CHANGES_REQUESTED** [review]

File: variables.tf (hunk 5 above)
Comment (ndegwajohn) [reply variables.tf:530]: This is the alias of the AWS managed AWS KMS for Amazon SNS

File: variables.tf (hunk 4 above)
Comment (ndegwajohn) [reply variables.tf:498]: The variable allows using an external sns topic that is not created using this module

**Review by hugoalmeida264 — APPROVED** [review]

**Pull request merged**